`azurerm_traffic_manager_endpoint` - supports for `min_child_endpoints_ipv4` and `min_child_endpoints_ipv6`

magodo opened the pull request:
This PR bumps API version of `trafficmanager` from `2018-04-01` to `2018-08-01`.

Also this PR adds support of the `min_child_endpoints_ipv4` and `min_child_endpoints_ipv6` for the `azurerm_traffic_manager_endpoint`.

katbyte commented on website/docs/r/traffic_manager_endpoint.html.markdown:
```diff
@@ -110,6 +110,20 @@ The following arguments are supported:
 
+~>**NOTE**: If `min_child_endpoints` is less than either `min_child_endpoints_ipv4` or `min_child_endpoints_ipv6`, then it won't have any effect.
+
+* `min_child_endpoints_ipv4` - (Optional) This argument specifies the minimum number
+    of IPv4 (DNS record type A) endpoints that must be ‘online’ in the child profile in order for the
+    parent profile to direct traffic to any of the endpoints in that child
+    profile. This argument only applies to Endpoints of type `nestedEndpoints`
+    and defaults to `1`.
+
+* `min_child_endpoints_ipv6` - (Optional) This argument specifies the minimum number
+    of IPv6 (DNS record type AAAA) endpoints that must be ‘online’ in the child profile in order for the
+    parent profile to direct traffic to any of the endpoints in that child
+    profile. This argument only applies to Endpoints of type `nestedEndpoints`
+    and defaults to `1`.
```
could we remove the line breaks here and format it normally?

katbyte commented on azurerm/internal/services/trafficmanager/traffic_manager_endpoint_resource.go:
```diff
@@ -116,6 +116,16 @@ func resourceArmTrafficManagerEndpoint() *pluginsdk.Resource {
 				Optional: true,
 			},
 
+			"min_child_endpoints_ipv4": {
+				Type:     pluginsdk.TypeInt,
+				Optional: true,
+			},
+
+			"min_child_endpoints_ipv6": {
```
could we rename these too
```suggestion
			"minimum_required_child_endpoints_ipv4": {
				Type:     pluginsdk.TypeInt,
				Optional: true,
			},

			"minimum_required_child_endpoints_ipv6": {
```

katbyte reviewed (changes requested):
just one comment but otherwise LGTM 👍

magodo commented:
@katbyte I've updated the document and rename the properties per your comments.

I didn't add the sanity check on the `min_child_endpoints` as it has to be larger than 0, which means I can't both add a check for `min_child_endpoints >= minimum_required_child_endpoints_ipv4 + minimum_required_child_endpoints_ipv6`, and also allows users to only specify either ipv4/ipv6 ones by only applying above check when `min_child_endpoints > 0`. Otherwise, if the `min_child_endpoints` is less than 1, the API will raise error. For this reason, I also added a validation for the `min_child_endpoints` to ensure it is larger than 0.

magodo commented:
The test failures are not related to this PR.

katbyte reviewed (approved):
LGTM 🚀
